### ISSUE TITLE
Add support for new classes in oe_get

### ISSUE THREAD

**Issue by agila5** [ITSLeeds/osmextract#97]
After #35 

- [x] sf
- [x] sfc_POLYGON
- [x] sfc_LINESTRING
- [ ] sp objects

**Comment by agila5** [comment]
See also https://github.com/ITSLeeds/osmextract/pull/129#discussion_r497319224 for spatial matching using something different from a `sfc_POINT`

**Comment by agila5** [comment]
I plan to work on `sfc_POLYGON` and `sfc_LINESTRING` extensions during the weekend (for a new project). The main question here is: how to determine the matched geographical area when we are comparing a POLYGON with a LINESTRING or a POLYGON with a POLYGON? The problem is that a POLYGON or a LINESTRING could intersect two or more geographical areas determined by a provider. 

The simplest solution should be "greatest overlapping area + a warning message", any other suggestion?

**Comment by Robinlovelace** [comment]
Good thinking @agila5. I'm confident that this code will handle all `sf`/`sfc` classes, finding all zones that *completely contain* the geographic object, regardless of whether that's a line, polygon or point cloud:

https://github.com/ITSLeeds/osmextract/blob/3b927dd66f006c00d63e12424614696af1fede63/R/gf_find.R#L35-L50

The code should automatically select the smallest area that matches to avoid it downloading the world after every geographic query!

**Comment by agila5** [comment]
Thanks, I will start from that code!

**Comment by Robinlovelace** [comment]
I'm happy to give this a go...

**Comment by Robinlovelace** [comment]
For now the following works : )

```
osm_luton = osmextract::oe_get(sf::st_centroid(region_luton))
```

**Comment by Robinlovelace** [comment]
Would be great for this to work for polygons also, just tried this:

```
buildings = osmextract::oe_get(study_area, layer = "multipolygons")
Error: At the moment there is no support for matching objects of class sf. Feel free to open a new issue at github.com/itsleeds/osmextract
```

**Comment by agila5** [comment]
> Would be great for this to work for polygons also, just tried this:

The easy solution would be to take the centroid and then fall back to the sfc_POINT method, is that reasonable?

**Comment by Robinlovelace** [comment]
> The easy solution would be to take the centroid and then fall back to the sfc_POINT method, is that reasonable?

I think even simpler than that would be an `st_contains()` that would work with any geometry type and also ensure that larger regions are selected, e.g. for roads that cross 2 regions. I'm up for giving it a bash.

**Comment by Robinlovelace** [comment]
Done I think! Don't need to support the legacy `sp` class system IMO.

**Comment by agila5** [comment]
Ok! Then, I will close :)
